README.md

```diff
@@ -29,3 +29,117 @@ The installer scripts also work fine without docker; just run the following two
+./vsdownload.py --dest ~/my_msvc/opt/msvc
+# Add wrapper scripts, do minor cleanup of the unpacked MSVC installation
+./install.sh ~/my_msvc/opt/msvc
+
+# Custom CMake
+git clone https://gitlab.kitware.com/mstorsjo/cmake
+git checkout 8c1957b8026c60d50a705555a3a212c9217c0a2e
+mkdir build 
+cd build
+../configure --prefix=~/my_msvc/opt/cmake --parallel=$(nproc) -- -DCMAKE_USE_OPENSSL=OFF
+make -j$(nproc)
+make install
+
+# Run wine at least once
+wineserver -k # kills server (optional)
+wineserver -p
+wine64 wineboot
+```
+
+### Setting up your project with CMake
+
+You need to set the path to prioritize our custom CMake, and also to see our MSVC installation
+After that we just run CMake command with a few extra settings:
+
+```bash
+export PATH=~/my_msvc/opt/cmake/bin:$PATH
+export PATH=~/my_msvc/opt/msvc/bin/x64:$PATH
+CC=cl CXX=cl cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_PROGRAMS=ON -DCMAKE_SYSTEM_NAME=Windows -DCMAKE_CROSSCOMPILING=ON
+make
+```
+
+# FAQ
+
+## Does it run on Ubuntu 18.04 LTS?
+
+Yes, but the install scripts won't work because `msitools` is too old. You'll need to install either via Docker or on a real Ubuntu 20.04 LTS machine; and later copy paste the files under `/opt/msvc`
+
+## Does it work with CMake?
+
+Yes, but a custom version is needed or else CMake will complain that `/opt/msvc/bin/x64/cl` can't build a simple program.
+
+It also fixes the `Ninja` Generator which otherwise has trouble finding RC.
+
+## Can it build Debug versions?
+
+Yes, but there may be troubles. From wine errors appearing in the logs/console to problems when trying to launch it because of missing debug redistributables.
+You will also have to install winbind (see next item).
+
+For the best out-of-the-box experience build in Release mode.
+
+## fatal error C1902: Program database manager mismatch; please check your installation
+
+You need winbind: `sudo apt install winbind`
+
+Issue is [being tracked](https://github.com/mstorsjo/msvc-wine/issues/6).
+
+## Does `Ninja` work?
+
+Yes.
+
+##  Does `Unix Makefile` work?
+
+Yes.
+
+## Project gets rebuild from scratch every time
+
+This is a [known issue](https://github.com/mstorsjo/msvc-wine/issues/29).
+Apply this patch:
+
+```diff
+diff --git a/wrappers/cl b/wrappers/cl
+index 06702ba..ced5360 100755
+--- a/wrappers/cl
++++ b/wrappers/cl
+@@ -1,3 +1,3 @@
+ #!/bin/bash
+ . $(dirname $0)/msvcenv.sh
+-$(dirname $0)/wine-msvc.sh $BINDIR/cl.exe "$@"
++$(dirname $0)/wine-msvc.sh $BINDIR/cl.exe "$@" | sed "/Note: including file:/ s/\\\\/\//g" -
+```
+
+## I get `ninja: error: build.ninja:225: bad $-escape (literal $ must be written as $$)`
+
+Visual Studio can switch between Debug/Release/RelWithDebInfo/etc at build time in the IDE.
+
+It's slightly common for CMake projects to use `$(CONFIGURATION)` macro from Visual Studio to resolve commands to each intended configuration automatically.
+
+However generators like `Ninja`/`Unix Makefile` can only target one configuration at a time.
```

Comment:
There's a `Ninja Multi-Config` generator available too, that should handle that setup.

Reply:
I didn't want to mention `Ninja Multi-Config` because I didn't test it. I don't know if it actually handles it correctly.